src/app/barter/services/transactions.service.ts

```diff
@@ -29,4 +32,24 @@ export class TransactionsService implements IFirebaseService<Transaction> {
   delete$(data: Transaction): Observable<void> {
     return this.firebaseService.deleteDocument<Transaction>(this.collectionName, data);
   }
+
+  getNewTransactions$(): Observable<Transaction[]> {
+    return this.newTransaction$.asObservable();
+  }
+
+  alertNewTransaction(transactions: Transaction[]) {
+    this.userStore$.select(getUser).subscribe((user: User) => {
+      const myNewTransactions: Transaction[] = transactions.filter(transaction =>
+        this.shouldFilterTransaction(!!transaction.transactionCompleteDate ? transaction.transactionCompleteDate.toDate() :
```

Comment:
Does it also filter notification by user? 
Example of what shouldnt happen - my user gets a notification - for a barter offer of shperling to you

Reply:
done